### DISCUSSION TITLE
Net::HTTPBadResponse: wrong header line format

### DISCUSSION THREAD

**Discussion by AlexITC** [postalserver/postal#2329] [Help with using Postal]
Hi, my team has been reporting this problem that is displayed at Postal UI -> Incoming Mails.

Is there any advice on how to dig further? my impression is that this could be an issue while Postal parses incoming email headers.

I'm attaching one of those emails.
[message.zip](https://github.com/postalserver/postal/files/10469302/message.zip)

**Comment by willpower232** [comment]
Where are you seeing this error message?

**Reply by AlexITC** [reply]
I figured that this is thrown by the underlying http library (see https://github.com/ruby/ruby/blob/master/lib/net/http/response.rb#L181) while parsing response headers.

> my impression is that this could be an issue while Postal parses incoming email headers

Let me correct, my new hypothesis is that there is a chance this could be caused while Postal invokes a webhook.

> Where are you seeing this error message?

This is shown by Postal UI at an incoming message.

![Screenshot from 2023-01-21 07-53-12](https://user-images.githubusercontent.com/4218436/213872439-67d16d85-786f-49e2-9156-c6b0e6560e94.png)

**Reply by AlexITC** [reply]
Digging more into this, it seems like one of our webhooks is returning status 500 while processing some requests from Postal, I'd bet that for some weird reason we are returning an invalid http header in the response which causes Postal to log this.

It would be nice if Postal included more details about the error, like the webhook that caused it.

**Reply by willpower232** [reply]
I think webhooks are separate, you might be able to rummage in the Postal SMTP log for the Support Ref and see more information. Are you delivering this message to an internal domain?

**Reply by AlexITC** [reply]
Unfortunately, I don't see anything in the logs, I picked up a recent ref id, then, I tried `sudo postal logs | grep NNKIA6LP` and `grep NNKIA6LP /opt/postal/log/*.log`

Fun fact, `sudo postal logs worker` does not display anything, do I need to enable these?

> Are you delivering this message to an internal domain?

I'm delivering this to one of our apps, which is the one crashing.

Thanks.

**Reply by willpower232** [reply]
There won't be any files to look in, I'm not sure if you need to actually look at `postal logs smtp` but if you have recently restarted then maybe that means the logs don't persist, I don't know for sure.

Do you mean you have an app which is receiving messages over SMTP from Postal but something in that app isn't quite right so its emitting bad header and breaking the delivery of the message?

**Reply by AlexITC** [reply]
> There won't be any files to look in,

Oh, I exposed Postal logs to a directory, still, i just see the ones for smtp and web, there is a `rails.log` which is empty, and, no `worker.log`.

> I'm not sure if you need to actually look at postal logs smtp but if you have recently restarted then maybe that means the logs don't persist, I don't know for sure.

I did indeed restarted postal but I triggered the error by re-processing an incoming mail, still, no log is displayed.

> Do you mean you have an app which is receiving messages over SMTP from Postal but something in that app isn't quite right so its emitting bad header and breaking the delivery of the message?

Postal invokes an http webhook with incoming emails,  we intend to process such incoming emails with our custom app, it seems there are some specific cases causing our app to fail processing the message, for some unknown reason, it seems to return an invalid http header in the error response.
